sdk/js/src/DefaultCacheStore.ts

```diff
@@ -32,5 +45,91 @@ const stubbedLocalStorage = {
     key: () => null,
     length: 0,
 }
+// IndexedDB implementation
+export class IndexedDBStrategy extends StorageStrategy {
+    override store: IDBDatabase
+    private ready: Promise<void>
+    private static storeName = 'DevcycleStore'
+    private static DBName = 'DevcycleDB'
+    constructor() {
+        super()
+        this.ready = new Promise((resolve, reject) => {
+            this.init()
+                .then((db) => {
+                    this.store = db
+                    resolve()
+                })
+                .catch((err) => reject(err))
+        })
+    }
+
+    async init(): Promise<IDBDatabase> {
+        return new Promise((resolve, reject) => {
+            const request = indexedDB.open(IndexedDBStrategy.DBName, 1)
+
+            request.onupgradeneeded = (event) => {
+                const db = request.result
+                if (
+                    !db.objectStoreNames.contains(IndexedDBStrategy.storeName)
+                ) {
+                    db.createObjectStore(IndexedDBStrategy.storeName, {
+                        keyPath: 'id',
+                    })
+                }
+            }
+
+            request.onsuccess = (event) => {
+                resolve(request.result)
+            }
 
-export default DefaultCacheStore
+            request.onerror = (event) => {
+                reject(request.error)
+            }
+        })
+    }
+    async save(storeKey: string, data: unknown): Promise<void> {
+        await this.ready
+        const tx = this.store.transaction(
+            IndexedDBStrategy.storeName,
+            'readwrite',
+        )
+        const store = tx.objectStore(IndexedDBStrategy.storeName)
+        store.put({ id: storeKey, data: data })
+    }
+
+    // IndexedDB load
+    async load<T>(storeKey: string): Promise<T | undefined> {
+        await this.ready
+        const tx = this.store.transaction(
+            IndexedDBStrategy.storeName,
+            'readonly',
+        )
+        const store = tx.objectStore(IndexedDBStrategy.storeName)
+        const request = store.get(storeKey)
+        return new Promise((resolve, reject) => {
+            request.onsuccess = () => {
+                resolve(request.result ? request.result.data : undefined)
+            }
+            request.onerror = () => reject(request.error)
+        })
+    }
+
+    // IndexedDB remove
+    async remove(storeKey: string): Promise<void> {
+        await this.ready
+        const tx = this.store.transaction(
+            IndexedDBStrategy.storeName,
+            'readwrite',
+        )
+        const store = tx.objectStore(IndexedDBStrategy.storeName)
+        store.delete(storeKey)
```

Comment:
can you await the delete here? or is it synchronous?

Reply:
store.get and store.delete are synchronous. The parent function only needs to be async for the `await this.ready`